src/battle_transition.c

```diff
@@ -831,402 +872,425 @@ static void HBCB_BT_Phase2HorizontalCorrugate(void)
 {
     DmaStop(0);
-    BT_VBSyncOamAndPltt();
-    if (sTransitionStructPtr->vblankDma)
+    VBlankCB_BattleTransition();
+    if (sTransitionData->vblankDma)
         DmaCopy16(3, gScanlineEffectRegBuffers[0], gScanlineEffectRegBuffers[1], 320);
-    SetGpuReg(REG_OFFSET_WININ, sTransitionStructPtr->winIn);
-    SetGpuReg(REG_OFFSET_WINOUT, sTransitionStructPtr->winOut);
-    SetGpuReg(REG_OFFSET_WIN0V, sTransitionStructPtr->win0V);
-    SetGpuReg(REG_OFFSET_BLDCNT, sTransitionStructPtr->bldCnt);
-    SetGpuReg(REG_OFFSET_BLDALPHA, sTransitionStructPtr->bldAlpha);
+    SetGpuReg(REG_OFFSET_WININ, sTransitionData->winIn);
+    SetGpuReg(REG_OFFSET_WINOUT, sTransitionData->winOut);
+    SetGpuReg(REG_OFFSET_WIN0V, sTransitionData->win0V);
+    SetGpuReg(REG_OFFSET_BLDCNT, sTransitionData->bldCnt);
+    SetGpuReg(REG_OFFSET_BLDALPHA, sTransitionData->bldAlpha);
 }
 
-static void VBCB_BT_Phase2BigPokeball1(void)
+static void VBlankCB_PatternWeave(void)
 {
-    BT_VBStopDma0SyncSrcBufferSetLcdRegs();
-    DmaSet(0, gScanlineEffectRegBuffers[1], &REG_BG0HOFS, ((DMA_ENABLE | DMA_START_HBLANK | DMA_REPEAT | DMA_16BIT | DMA_SRC_INC | DMA_DEST_FIXED) << 16) | 1);
+    VBlankCB_SetWinAndBlend();
+    DmaSet(0, gScanlineEffectRegBuffers[1], &REG_BG0HOFS, B_TRANS_DMA_FLAGS);
 }
 
-static void VBCB_BT_Phase2BigPokeball2(void)
+static void VBlankCB_CircularMask(void)
 {
-    BT_VBStopDma0SyncSrcBufferSetLcdRegs();
-    DmaSet(0, gScanlineEffectRegBuffers[1], &REG_WIN0H, ((DMA_ENABLE | DMA_START_HBLANK | DMA_REPEAT | DMA_16BIT | DMA_SRC_INC | DMA_DEST_FIXED) << 16) | 1);
+    VBlankCB_SetWinAndBlend();
+    DmaSet(0, gScanlineEffectRegBuffers[1], &REG_WIN0H, B_TRANS_DMA_FLAGS);
 }
 
-#undef tRadius
-#undef tDeltaRadius
-#undef tKeepVBCB
-#undef tTheta
 #undef tAmplitude
+#undef tSinIndex
+#undef tBlendTarget1
+#undef tBlendTarget2
+#undef tRadius
+#undef tRadiusDelta
+#undef tVBlankSet
+
+//------------------------------
+// B_TRANSITION_POKEBALLS_TRAIL
+//------------------------------
+
+#define sSide  data[0]
+#define sDelay data[1]
+#define sPrevX data[2]
 
-// TODO: Document this effect after knowing more about field effects.
-static void BT_Phase2SlidingPokeballs(u8 taskId)
+static void Task_PokeballsTrail(u8 taskId)
 {
-    while (sBT_Phase2SlidingPokeballsFuncs[gTasks[taskId].tState](&gTasks[taskId]));
+    while (sPokeballsTrail_Funcs[gTasks[taskId].tState](&gTasks[taskId]));
 }
 
-static bool8 BT_Phase2SlidingPokeballs_LoadBgGfx(struct Task *task)
+static bool8 PokeballsTrail_Init(struct Task *task)
 {
-    u16 *tilemapAddr, *tilesetAddr;
+    u16 *tilemap, *tileset;
 
-    BT_GetBg0TilemapAndTilesetBase(&tilemapAddr, &tilesetAddr);
-    CpuCopy16(sSlidingPokeballTilemap, tilesetAddr, 0x40);
-    CpuFill32(0, tilemapAddr, 0x800);
-    LoadPalette(sSlidingPokeballBigPokeballPalette, 0xF0, 0x20);
-    ++task->tState;
+    GetBg0TilesDst(&tilemap, &tileset);
+    CpuCopy16(sSlidingPokeball_Tilemap, tileset, 0x40);
```

Comment:
use sizeof? There are also other cases where the size is hardcoded

Reply:
I'm sure, these repos are filled with missing `sizeof` usage. I'm not going to root around for them now, the purpose of this PR is just to bring it up to speed with pokeemerald.